### PR TITLE
feat(test): start authoring raw reporter

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

**Review by dgozman — APPROVED** [review]

File: src/test/reporters/raw.ts [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+  async onEnd() {
+    const projectSuites = this.suite.suites;
+    for (const suite of projectSuites) {
+      const project = (suite as any)._projectConfig as FullProject;
+      assert(project, 'Internal Error: Invalid project structure');
+      const reportFolder = path.join(project.outputDir, 'report');
+      fs.mkdirSync(reportFolder, { recursive: true });
+      let reportFile: string | undefined;
+      for (let i = 0; i < 10; ++i) {
+        reportFile = path.join(reportFolder, sanitizeForFilePath(project.name || 'project') + (i ? '-' + i : '') + '.report');
```
Comment (dgozman) [inline src/test/reporters/raw.ts:121]: Maybe keep '.json'. for tooling, e.g. opening in IDE?

File: src/test/cli.ts [hunk 2]
```diff
@@ -81,9 +83,32 @@ export function addTestCommand(program: commander.CommanderStatic) {
   });
 }
 
-async function runTests(args: string[], opts: { [key: string]: any }) {
-  await startProfiling();
+export function addGenerateHtmlCommand(program: commander.CommanderStatic) {
+  const command = program.command('generate-html');
```
Comment (dgozman) [inline src/test/cli.ts:87]: Let's do this under some env.PWTEST_HTML_SUPPORT for now?

File: src/test/cli.ts [hunk 3]
```diff
@@ -81,9 +83,32 @@ export function addTestCommand(program: commander.CommanderStatic) {
...
+  command.description('Generate HTML report');
+  command.option('-c, --config <file>', `Configuration file, or a test directory with optional "${tsConfig}"/"${jsConfig}"`);
+  command.option('--output <dir>', `Folder for output artifacts (default: "test-results")`);
+  command.option('--project <project-name...>', `Only run tests from the specified list of projects (default: run all projects)`);
```
Comment (dgozman) [inline src/test/cli.ts:91]: Do we need this one?  If we do, let's update the description.

File: src/test/cli.ts [hunk 4]
```diff
@@ -81,9 +83,32 @@ export function addTestCommand(program: commander.CommanderStatic) {
   });
 }
 
-async function runTests(args: string[], opts: { [key: string]: any }) {
-  await startProfiling();
+export function addGenerateHtmlCommand(program: commander.CommanderStatic) {
+  const command = program.command('generate-html');
+  command.description('Generate HTML report');
+  command.option('-c, --config <file>', `Configuration file, or a test directory with optional "${tsConfig}"/"${jsConfig}"`);
+  command.option('--output <dir>', `Folder for output artifacts (default: "test-results")`);
```
Comment (dgozman) [inline src/test/cli.ts:90]: I don't see this one being used, is it?

**Review by dgozman — APPROVED** [review]

File: src/test/reporters/raw.ts (hunk 1 above)
Comment (pavelfeldman) [reply src/test/reporters/raw.ts:121]: I wouldn't want it to be opened in vscode. Will probably use cbor to defend myself :)

**Pull request merged**